src/csharp/Grpc.Core.Tests/CompressionTest.cs

```diff
@@ -118,5 +119,30 @@ public void WriteOptions_Unary()
 
             await call.ResponseStream.ToListAsync();
         }
+
+        [Test]
+        public void CanReadCompressedMessages()
+        {
+            var compressionMetadata = new Metadata
+            {
+                { new Metadata.Entry("grpc-internal-encoding-request", "gzip") }
```

Comment:
not directly. as it's a C macro. I would need to define a constant with the same value somewhere, but I am not sure I want to expose it publicly. This way of setting the compression doesn't seem to be something I'd recommend to the users, I mostly wanted to test that it's possible to set the compression when needed and that the compression disabling on streams works as expected.